packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts

```diff
@@ -104,13 +106,76 @@ export class RemoteTableService {
       throw new NotFoundException('Remote server does not exist');
     }
 
-    const remoteTable = await this.createForeignTableAndMetadata(
+    if (!input.schema) {
+      throw new BadRequestException(
+        'Schema is required for syncing remote table',
+      );
+    }
+
+    const currentRemoteTableWithSameDistantName =
+      await this.remoteTableRepository.findOne({
+        where: {
+          distantTableName: input.name,
+          remoteServerId: remoteServer.id,
+          workspaceId,
+        },
+      });
+
+    if (currentRemoteTableWithSameDistantName) {
+      throw new BadRequestException('Remote table already exists');
+    }
+
+    const dataSourceMetatada =
+      await this.dataSourceService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
+        workspaceId,
+      );
+
+    const localTableName = getRemoteTableLocalName(input.name);
+
+    this.validateTableNameDoesNotExists(
+      localTableName,
+      workspaceId,
+      dataSourceMetatada.schema,
+    );
+
+    const remoteTableEntity = this.remoteTableRepository.create({
```

Comment:
same here?

Reply:
this is not an async function this one :)